controllers/job.go

```diff
@@ -181,13 +185,34 @@ func (controller *JobController) Post(w http.ResponseWriter, r *http.Request, pa
 		return
 	}
 
-	updateData := struct{ NextState string }{}
+	updateData := struct {
+		NextState string
+		Timeout   time.Duration
+	}{}
 	err := json.NewDecoder(r.Body).Decode(&updateData)
 	if err != nil {
 		writeErr(w, err)
 		return
 	}
 
+	new_timeout := false
+	if updateData.Timeout != 0 {
+		new_timeout = true
+		job.Timeout = updateData.Timeout
+	}
+
+	if updatingTimeoutForNonInflightState := new_timeout &&
+		updateData.NextState != data.JobInflight.String(); updatingTimeoutForNonInflightState {
+		writeStatus(w, http.StatusBadRequest, errTimeoutWithInvalidState)
+		return
+	}
+	if updatingTimeoutForAlreadyInflightState := new_timeout &&
+		job.Status == data.JobInflight &&
+		job.Status.String() == updateData.NextState; updatingTimeoutForAlreadyInflightState {
+		writeStatus(w, http.StatusBadRequest, errTimeoutWithInvalidState)
+		return
+	}
```

Comment:
```suggestion
	if updateData.Timeout != 0 {
		if updateData.NextState != data.JobInflight.String() || (job.Status == data.JobInflight && job.Status.String() == updateData.NextState) {
			writeStatus(w, http.StatusBadRequest, errTimeoutWithInvalidState)
			return
		}

		job.Timeout = updateData.Timeout
	}
```

Would this be simpler?

Reply:
Good suggestion @RedwanPlague , I found this code complex, but didn't realize it was doing the exact same thing

Reply:
Thank you @RedwanPlague!